runtime/resolvers/metrics.go

```diff
@@ -142,9 +146,42 @@ func (r *metricsResolver) ResolveInteractive(ctx context.Context) (runtime.Resol
 	if err != nil {
 		return nil, err
 	}
+
+	if r.meta != nil {
+		// If metadata is provided, we create a result with the metadata
+		return runtime.NewDriverResolverWithMetaResult(res, r.meta), nil
+	}
+
 	return runtime.NewDriverResolverResult(res), nil
 }
 
 func (r *metricsResolver) ResolveExport(ctx context.Context, w io.Writer, opts *runtime.ResolverExportOptions) error {
 	return errors.New("not implemented")
 }
+
+// buildMetricsViewMeta builds metadata for the measures and dimensions defined in the MetricsViewSpec.
+func buildMetricsViewMeta(spec *runtimev1.MetricsViewSpec) []map[string]any {
```

Comment:
This just returns a description of all the fields in the metrics view. It needs to return a description of only the fields in the result (i.e. the fields selected in `r.query` and returned from `r.executor.Query`).